packages/mattermost-redux/src/selectors/entities/search.ts

```diff
@@ -26,3 +27,28 @@ export const getAllUserMentionKeys: (state: GlobalState) => UserMentionKey[] = c
         return userMentionKeys.concat(groupMentionKeys);
     },
 );
+
+export function recentSearches(state: GlobalState): SearchParams[] {
+    const recentSearches = state.entities.search.recentSearches;
+    const channels = state.entities.channels.channels;
+    const userProfiles = state.entities.users.profiles;
+    if (!recentSearches) {
+        return [];
+    }
+    return recentSearches.map((searchParams) =>
```

Comment:
Since this returns a newly constructed array, it has to be memoized or else it'll return a new array every time it's called which would cause the component using it to re-render on every store change.

You should be able to just do that by using `createSelector` and getting the data from state in the input selectors before doing this mapping in the final function

Reply:
Thanks for the explanation! I will refactor it to user the `createSelector`.